GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -256,36 +260,31 @@ public bool IsReadyForExternalAcquireLockRequests(NamedPipeMessages.LockData req
             return true;
         }
 
-        public EventMetadata GetMetadataForHeartBeat(ref EventLevel eventLevel)
+        public EventMetadata GetAndResetHeartBeatMetadata(out bool logToFile)
         {
+            logToFile = false;
             EventMetadata metadata = new EventMetadata();
-            if (this.placeHolderCreationCount.Count > 0)
-            {
-                ConcurrentDictionary<string, PlaceHolderCreateCounter> collectedData = this.placeHolderCreationCount;
-                this.placeHolderCreationCount = new ConcurrentDictionary<string, PlaceHolderCreateCounter>(StringComparer.OrdinalIgnoreCase);
-
-                int count = 0;
-                foreach (KeyValuePair<string, PlaceHolderCreateCounter> processCount in
-                    collectedData.OrderByDescending((KeyValuePair<string, PlaceHolderCreateCounter> kvp) => kvp.Value.Count))
-                {
-                    ++count;
-                    if (count > 10)
-                    {
-                        break;
-                    }
-
-                    metadata.Add("ProcessName" + count, processCount.Key);
-                    metadata.Add("ProcessCount" + count, processCount.Value.Count);
-                }
 
-                eventLevel = EventLevel.Informational;
-            }
+            ConcurrentDictionary<string, PlaceHolderCreateCounter> storedProcessData = this.GetAndResetProcessCountMetadata(ref this.filePlaceHolderCreationCount);

```

Comment:
nit: I find the use of storedProcessData a bit hard to follow.  Can we avoid this variable and just do

`this.GetProcessInteractionData(this.GetAndResetProcessCountMetadata(ref this.filePlaceHolderCreationCount), ref logToFile));`

Same comment for other two instances